blockchain/chain/src/store/chain_store.rs

```diff
@@ -91,19 +95,22 @@ pub struct ChainStore<DB> {
     /// Tipset at the head of the best-known chain.
     heaviest: RwLock<Option<Arc<Tipset>>>,
 
+    ts_cache: RwLock<LruCache<TipsetKeys, Arc<Tipset>>>,
+
     /// tip_index tracks tipsets by epoch/parentset for use by expected consensus.
-    tip_index: RwLock<TipIndex>,
+    tip_index: TipIndex,
```

Comment:
The PR description says we're removing TipIndex? Why u no remove tipindex? Is this what you mean by half of #814 ?

Reply:
I just mean I gutted the TipIndex functionality and usages of it, I decided to keep for now because that will be built upon in next PR (why remove when it will just be re-added soon?)